src/main/java/com/yieldlab/gdpr/GdprConsent.java

```diff
@@ -0,0 +1,806 @@
+package com.yieldlab.gdpr;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.Base64.Encoder;
+import java.util.List;
+import java.util.Objects;
+
+import com.yieldlab.gdpr.exception.GdprException;
+import com.yieldlab.gdpr.exception.VendorConsentCreateException;
+import com.yieldlab.gdpr.exception.VendorConsentException;
+import com.yieldlab.gdpr.exception.VendorConsentParseException;
+
+/**
+ * Copied from https://github.com/InteractiveAdvertisingBureau/Consent-String-SDK-Java and modified
+ *
+ * This class implements a parser for the IAB consent string as specified in
+ * https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/
+ * Draft_for_Public_Comment_Transparency%20%26%20Consent%20Framework%20-%20cookie%20and%20vendor%20list%20format%
+ * 20specification%20v1.0a.pdf
+ *
+ */
+public class GdprConsent {
```

Comment:
Why did you rename the class?

Reply:
I think it makes sense to separate the parsing logic into its own class and leave this one as a pure immutable value class?

Reply:
I have restored the name of the class on popular request.